app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+    # Enforces unique not null constraint on username and email
+    validates :username, presence: true,
+              uniqueness: { case_sensitive: false },
+              length: {minimum: 3, maximum: 25}
+
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+    validates :email, presence: true,
+              uniqueness: { case_sensitive: false },
+              length: { maximum: 105 },
+              format: { with: VALID_EMAIL_REGEX }
+
+    has_secure_password
+end
```

Comment:
Can I check if we are also creating the `uid` column, or are we leaving it to rails to generate it as part of the table?

Reply:
I think we just leave it to rails to generate it as part of the table.